compiler/parser/RdrHsSyn.hs

```diff
@@ -2549,20 +2550,26 @@ extractTemplateBodyDecls = foldl extract ([], [], [], [], [], [], [])
         LetBindingsDecl (L _ (_, b)) -> (es, ss, os, as, gs, b : bs, fs)
         FlexibleChoiceDecl f         -> (es, ss, os, as, gs, bs, f : fs)
 
+-- | Utility for calculating 'DA.Internal.Desugar' names referenced
+-- during desugaring.
+desugarRdrName :: (String -> OccName) -> String -> RdrName
```

Comment:
This function is odd. It does one function composition more than makes sense. Why not:

```
qualifyDesugar :: OccName -> RdrName
qualifyDesugar = mkRdrQual (mkModuleName "DA.Internal.Desugar")
```

All the users gain one `$` but I think it's a big win for separation of concerns and clarity.